File: OpenTokSamples/app/src/main/java/com/opentok/android/demo/screensharing/ScreensharingCapturer.java
package com.opentok.android.demo.screensharing;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.os.Handler;
import android.view.View;

import com.opentok.android.BaseVideoCapturer;

public class ScreensharingCapturer extends BaseVideoCapturer {
    boolean capturing = false;
    Context context;
    View contentView;

    int fps = 15;
    int width = 0;
    int height = 0;
    int[] frame;

    Bitmap bmp;
    Canvas canvas;

    Handler mHandler = new Handler();

    Runnable newFrame = new Runnable() {
        @Override
        public void run() {
            if (capturing) {
                int width = contentView.getWidth();
                int height = contentView.getHeight();

                if (frame == null ||
                        ScreensharingCapturer.this.width != width ||
                        ScreensharingCapturer.this.height != height) {

                    ScreensharingCapturer.this.width = width;
                    ScreensharingCapturer.this.height = height;

                    if (bmp != null) {
                        bmp.recycle();
                        bmp = null;
                    }

                    bmp = Bitmap.createBitmap(width,
                            height, Bitmap.Config.ARGB_8888);

                    canvas = new Canvas(bmp);
                    frame = new int[width * height];
                }

                contentView.draw(canvas);

                bmp.getPixels(frame, 0, width, 0, 0, width, height);

                provideIntArrayFrame(frame, ARGB, width, height, 0, false);

                mHandler.postDelayed(newFrame, 1000 / fps);

            }
        }
    };

    public ScreensharingCapturer(Context context, View view) {
        this.context = context;
        this.contentView = view;
    }

    @Override
    public void init() {

    }

    @Override
    public int startCapture() {
        capturing = true;

        mHandler.postDelayed(newFrame, 1000 / fps);
        return 0;
    }

    @Override
    public int stopCapture() {
        capturing = false;
        mHandler.removeCallbacks(newFrame);
        return 0;
    }

    @Override
    public boolean isCaptureStarted() {
        return capturing;
    }

    @Override
    public CaptureSettings getCaptureSettings() {

        CaptureSettings settings = new CaptureSettings();
        settings.fps = fps;
        settings.width = width;
        settings.height = height;
        settings.format = ARGB;
        return settings;
    }

    @Override
    public void destroy() {

    }

    @Override
    public void onPause() {

    }

    @Override
    public void onResume() {

    }

}

File: Webview-Screen-Sharing-Kotlin/app/src/main/java/com/tokbox/sample/webviewscreensharing/ScreenSharingCapturer.kt
package com.tokbox.sample.webviewscreensharing

import android.content.Context
import android.graphics.Bitmap
import android.graphics.Canvas
import android.os.Handler
import android.view.View
import com.opentok.android.BaseVideoCapturer
import java.util.concurrent.atomic.AtomicBoolean

class ScreenSharingCapturer(private val context: Context, private val contentView: View) :
    BaseVideoCapturer() {
    private val capturing = AtomicBoolean(false)
    private val fps = 15
    private var width = 0
    private var height = 0
    private var frame: IntArray? = null
    private var bmp: Bitmap? = null
    private var canvas: Canvas? = null
    private val handler = Handler()
    private val newFrame: Runnable = object : Runnable {
        override fun run() {
            if (capturing.get()) {
                val width = contentView.width
                val height = contentView.height
                if (frame == null || this@ScreenSharingCapturer.width != width || this@ScreenSharingCapturer.height != height) {
                    this@ScreenSharingCapturer.width = width
                    this@ScreenSharingCapturer.height = height
                    if (bmp != null) {
                        bmp!!.recycle()
                        bmp = null
                    }
                    bmp = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
                    canvas = Canvas(bmp!!)
                    frame = IntArray(width * height)
                }

                val localCanvas = canvas
                if (localCanvas != null) {
                    localCanvas.saveLayer(0f, 0f, width.toFloat(), height.toFloat(), null)
                    localCanvas.translate(-contentView.scrollX.toFloat(), -contentView.scrollY.toFloat())
                    contentView.draw(localCanvas)
                    val localFrame = frame
                    if(localFrame != null){
                        bmp!!.getPixels(localFrame, 0, width, 0, 0, width, height)
                        provideIntArrayFrame(frame, ARGB, width, height, 0, false)
                        localCanvas.restore()
                        handler.postDelayed(this, (1000 / fps).toLong())
                    }
                }

            }
        }
    }

    override fun init() {}
    override fun startCapture(): Int {
        if (capturing.compareAndSet(false, true)) {
            // Start capturing
            handler.postDelayed(newFrame, (1000 / fps).toLong())
            return 0
        } else {
            // Already capturing
            return -1
        }
    }

    override fun stopCapture(): Int {
        if (capturing.compareAndSet(true, false)) {
            // Stop capturing
            handler.removeCallbacks(newFrame)
            return 0
        } else {
            // Not capturing
            return -1
        }
    }

    override fun isCaptureStarted(): Boolean {
        return capturing.get()
    }

    override fun getCaptureSettings(): CaptureSettings {
        val captureSettings = CaptureSettings()
        captureSettings.fps = fps
        captureSettings.width = width
        captureSettings.height = height
        captureSettings.format = ARGB
        return captureSettings
    }

    override fun destroy() {}
    override fun onPause() {}
    override fun onResume() {}
}